luigi/contrib/s3.py

```diff
@@ -340,8 +345,19 @@ def copy(self, source_path, destination_path, threads=100, start_time=None, end_
                         'Key': src_prefix + path
                     }
 
-                    self.s3.meta.client.copy(
-                        copy_source, dst_bucket, dst_prefix + path, Config=transfer_config, ExtraArgs=kwargs)
+                    the_kwargs = {'Config': transfer_config, 'ExtraArgs': kwargs}
+                    job = management_pool.apply_async(self.s3.meta.client.copy,
+                                                      args=(copy_source, dst_bucket, dst_prefix + path),
+                                                      kwds=the_kwargs)
+                    copy_jobs.append(job)
+
+            # Wait for the pools to finish scheduling all the copies
+            management_pool.close()
+            management_pool.join()
```

Comment:
Please just check that this doesn't have a ContextManager interface maybe?

Reply:
A few minutes of Google searching doesn't return valuable resources towards this desire.